scripts/prepare_sector_network.py

```diff
@@ -2917,27 +2918,42 @@ def add_electricity_distribution_grid(n, costs):
         )
 
 
-# def add_co2limit(n, Nyears=1.0, limit=0.0):
-#     print("Adding CO2 budget limit as per unit of 1990 levels of", limit)
-
-#     countries = n.buses.country.dropna().unique()
-
-#     sectors = emission_sectors_from_opts(opts)
-
-#     # convert Mt to tCO2
-#     co2_totals = 1e6 * pd.read_csv(snakemake.input.co2_totals_name, index_col=0)
-
-#     co2_limit = co2_totals.loc[countries, sectors].sum().sum()
+def add_co2_budget(n, co2_budget, investment_year, elec_opts):
+    # Check if CO2Limit already exists
+    if "CO2Limit" in n.global_constraints.index:
+        if co2_budget["overwrite"]:

```

Comment:
I wonder if it would increase readability if nested `if`-s would be replaced by `and` condition? According to [pythonic zen](https://en.wikipedia.org/wiki/Zen_of_Python) `Flat is better than nested`